x-pack/plugins/ml/public/ui_actions/edit_swimlane_panel_action.tsx

```diff
@@ -39,20 +45,23 @@ export function createEditSwimlanePanelAction(
           '../embeddables/anomaly_swimlane/anomaly_swimlane_setup_flyout'
         );
 
-        const result = await resolveAnomalySwimlaneUserInput(
-          coreStart,
-          deps.data.dataViews,
-          embeddable.getInput()
-        );
-        embeddable.updateInput(result);
+        const { jobIds, viewBy, swimlaneType, panelTitle } = context.embeddable;
+
+        const result = await resolveAnomalySwimlaneUserInput(coreStart, deps.data.dataViews, {
+          jobIds,
```

Comment:
If I edit the selected job(s), apply the edit (but don't save the Dashboard changes) and then hit edit again, it isn't correctly setting the selected job IDs in the flyout but is still using the previous selection.

<img width="1519" alt="Screenshot 2024-03-21 at 10 41 00" src="https://github.com/elastic/kibana/assets/7405507/1ae2ce54-5c87-495a-bf03-0318592b4eaa">

Reply:
Fixed in https://github.com/elastic/kibana/pull/179073/commits/9be954f9d31071a529f689ed71cd783dc3b70907